src/document-click.js

```diff
@@ -262,17 +274,25 @@ export function onDocumentElementClick_(
   // The choice of `location.replace` vs `history.replaceState` is important.
   // Due to bugs, not every browser triggers `:target` pseudo-class when
   // `replaceState` is called. See http://www.zachleat.com/web/moving-target/
-  // for more details.
-  win.location.replace(`#${hash}`);
+  // for more details. Do this only if fragment has changed.
+  if (tgtLoc.hash != curLoc.hash) {
+    win.location.replace(`#${hash}`);
+  }
 
   // Scroll to the element if found.
   if (elem) {
     viewport./*OK*/scrollIntoView(elem);
+    elem.setAttribute(oldAttribute.key, oldAttribute.value);
   } else {
     dev().warn('documentElement',
         `failed to find element with id=${hash} or a[name=${hash}]`);
   }
 
+  // Has fragment really changed?
+  if (tgtLoc.hash != curLoc.hash) {
```

Comment:
It should be `==` here I believe. Also, please remove `return` and wrap the `history.push` inside the `if`.